src/main/java/appeng/core/sync/packets/PacketMatterCannon.java

```diff
@@ -20,84 +20,60 @@
 
 
 import io.netty.buffer.ByteBuf;
-import io.netty.buffer.Unpooled;
 
 import net.minecraft.client.Minecraft;
-import net.minecraft.entity.player.EntityPlayer;
 import net.minecraft.init.Items;
 import net.minecraft.world.World;
 
 import cpw.mods.fml.client.FMLClientHandler;
-import cpw.mods.fml.relauncher.Side;
-import cpw.mods.fml.relauncher.SideOnly;
+import cpw.mods.fml.common.network.simpleimpl.MessageContext;
 
 import appeng.client.render.effects.MatterCannonFX;
 import appeng.core.sync.AppEngPacket;
-import appeng.core.sync.network.INetworkInfo;
+import appeng.core.sync.AppEngPacketHandler;
 
 
-public class PacketMatterCannon extends AppEngPacket
+public class PacketMatterCannon implements AppEngPacket, AppEngPacketHandler<PacketMatterCannon, AppEngPacket>
 {
 
-	final double x;
-	final double y;
-	final double z;
-	final double dx;
-	final double dy;
-	final double dz;
-	final byte len;
+	private float x;
+	private float y;
+	private float z;
+	private float dx;
+	private float dy;
+	private float dz;
+	private byte len;
 
 	// automatic.
-	public PacketMatterCannon( ByteBuf stream )
+	public PacketMatterCannon()
```

Comment:
![MAJOR](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-major.png) Add a nested comment explaining why this method is empty, throw an UnsupportedOperationException or complete the implementation. [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](http://nemo.sonarqube.org/coding_rules#rule_key=squid%3AS1186)